alexgburnet/Alex-Ethan-Sonali-Hannah-SEPP-Project
# Ci cd/docker fix

## What part of the codebase does this PR concern



## What type of PR is this? (check all applicable)

- [ ] Adding Testing
- [x] Adding CI/CD

## Description

- Moved .env.template and .gitignore to root to be in the right location for the docker-compose to work
- Added docker-compose, which starts the postgres dockerfile, passing in variables for database user, password, name
- Adjusted deploy workflow to write variables to the new location (root) on the vm

## To run:

- Ensure you have docker installed **and** running

- in the root directory, duplicate .env.template and rename the duplicate '.env'

- inside '.env' place the desired details for the database

- in the root of the directory still, run:
```bash
docker-compose down -v  # This will shut down the current processes, removing any persistent storage

docker compose up --build -V # This will restart it again
```

- To test it is working, in a new terminal run:
```bash
psql -h localhost -U <.env username> -d <.env database name>
```

- you will be prompted to type in the password from the .env

> Note: Make sure you do not have psql running as a service on your local machine when you try this as that will likely have higher precedence than the docker container
